### PR TITLE
part 2 was implemented

### PR DESCRIPTION
**Pull request by Mandybur800**

### CODE REVIEW

File: src/main/java/ua/com/conductor/dao/GenericDao.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package ua.com.conductor.dao;
+
+import java.util.List;
+
+public interface GenericDao<T, I> {
```
Comment (Max-Pochepets) [inline src/main/java/ua/com/conductor/dao/GenericDao.java]: You have [functionality](https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_04.html) that is redundant at this state of project, I will list it below. So, for now, let's not use `Generic` interfaces. In fact, you won't be able to use them, or will be forced to change your corresponding interfaces

File: src/main/java/ua/com/conductor/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getAllSessionsDateQuery =
+                    session.createQuery("SELECT m FROM MovieSession m "
+                    + " JOIN FETCH m.cinemaHall JOIN FETCH m.movie"
+                            + " WHERE m.showTime >= :start AND m.showTime < :end ",
+                            MovieSession.class);
```
Comment (Max-Pochepets) [inline src/main/java/ua/com/conductor/dao/impl/MovieSessionDaoImpl.java]: Test a situation, where some fields of this entity won't be set

File: src/main/java/ua/com/conductor/dao/impl/MovieSessionDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            LocalDateTime start = date.atStartOfDay();
+            LocalDateTime end = date.plusDays(1).atStartOfDay();
```
Comment (Max-Pochepets) [inline src/main/java/ua/com/conductor/dao/impl/MovieSessionDaoImpl.java]: You can simplify this by SQL Date functions. FYI

File: src/main/java/ua/com/conductor/dao/impl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    public List<MovieSession> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getAllSessionsQuery =
+                    session.createQuery("SELECT m FROM MovieSession m "
+                    + " JOIN FETCH m.cinemaHall JOIN FETCH m.movie", MovieSession.class);
+            return getAllSessionsQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get all movie sessions ", e);
+        }
+    }
```
Comment (Max-Pochepets) [inline src/main/java/ua/com/conductor/dao/impl/MovieSessionDaoImpl.java]: You don't need it in the scope of this task

File: src/main/java/ua/com/conductor/service/impl/MovieSessionServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        return movieSessionDao.add(movieSession);
+    }
+
+    @Override
+    public List<MovieSession> getAll() {
+        return movieSessionDao.getAll();
+    }
```
Comment (Max-Pochepets) [inline src/main/java/ua/com/conductor/service/impl/MovieSessionServiceImpl.java]: same

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]
👍

File: src/main/java/ua/com/conductor/Application.java [hunk 6]
```diff
@@ -12,6 +18,39 @@ public static void main(String[] args) {
...
+        movieSessionThree.setMovie(movieSimon);
+        movieSessionThree.setCinemaHall(cinemaHallTwo);
+        movieSessionThree.setShowTime(LocalDateTime.of(2021, 1, 20, 20, 0));
+        MovieSessionService movieSessionService = (MovieSessionService)
+                injector.getInstance(MovieSessionService.class);
+        movieSessionService.add(movieSessionOne);
+        movieSessionService.add(movieSessionTwo);
+        movieSessionService.add(movieSessionThree);
+        System.out.println();
+        movieSessionService.findAvailableSessions(1L, LocalDate.now())
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/conductor/Application.java]: let's use `movie.getId()` instead

File: src/main/java/ua/com/conductor/model/MovieSession.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+@Entity
+@Table(name = "movie_session")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Movie movie;
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "cinema_hall")
```
Comment (KseniiaMakarova) [inline src/main/java/ua/com/conductor/model/MovieSession.java]:
```suggestion
    @JoinColumn(name = "cinema_hall_id")
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
let's fix travis

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**